### PR TITLE
Remove SystemNative_GetUnixName() PAL function

### PR DESCRIPTION
**Pull request by akoeplinger**
After working on https://github.com/dotnet/runtime/pull/68313 I realized we don't need this PAL function at all since we can hardcode all existing operating systems in the managed layer (if you port to a new OS you need to port System.Private.CoreLib anyway).

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/OperatingSystem.cs [hunk 1]
```diff
@@ -103,8 +99,16 @@ public static bool IsOSPlatform(string platform!!)
...
+#elif TARGET_FREEBSD
+            return platform.Equals("FREEBSD", StringComparison.OrdinalIgnoreCase);
+#elif TARGET_NETBSD
+            return platform.Equals("NETBSD", StringComparison.OrdinalIgnoreCase);
+#elif TARGET_ILLUMOS
+            return platform.Equals("ILLUMOS", StringComparison.OrdinalIgnoreCase);
+#elif TARGET_SOLARIS
+            return platform.Equals("SOLARIS", StringComparison.OrdinalIgnoreCase);
 #else
 #error Unknown OS
```
Comment (am11) [inline src/libraries/System.Private.CoreLib/src/System/OperatingSystem.cs:113]: nit: We are deleting this build-time error message from cmake:

```cmake
message(FATAL_ERROR "Unknown platform. Cannot define PAL_UNIX_NAME, used by RuntimeInformation.")
```

which is slightly more helpful. Can we use similar message here?

**Comment by am11** [comment]
> if you port to a new OS you need to port System.Private.CoreLib anyway

Makes sense. Also it reduces number of files to modify for this, during the port of new platform.

File: src/libraries/System.Private.CoreLib/src/System/OperatingSystem.cs [hunk 2]
```diff
@@ -103,8 +99,16 @@ public static bool IsOSPlatform(string platform!!)
             return platform.Equals("TVOS", StringComparison.OrdinalIgnoreCase);
 #elif TARGET_ANDROID
```
Comment (jkotas) [inline src/libraries/System.Private.CoreLib/src/System/OperatingSystem.cs:100]: Can we define `const string OSPlatformName = ` using the ifdefs and just use it here to make this less repetitive?

Comment (akoeplinger) [reply src/libraries/System.Private.CoreLib/src/System/OperatingSystem.cs]: done

**Review by jkotas — APPROVED** [review]
Thanks!

**Pull request merged**